### PR TITLE
Create edit input validation

### PR DESCRIPTION
**Pull request by flesher1**
This adds validation to submission and saving. 

Submission has the same rules as saving, but saving has no required fields meaning that reuse of submission validation cannot be done. 

To test:
create a new listing. attempt to save for every field. 
once complete, submit. 
also test a listing with only submits and no saving. 
test to make sure existing listings load and can be modified (excluding tags)

### CODE REVIEW

File: src/components/createEdit/sections/OwnerContact.js [hunk 1]
```diff
@@ -35,23 +36,28 @@ class POCForm extends React.Component{
...
                     <div>
                         Name
                         <input autoComplete='name' id={"name"} value={this.props.value.name || ''} className="form-control" required type="text" dir="auto" onChange={this.props.handleChange}/>
+                        <div className="errorText">{this.props.error.name? "Required" : ""}</div>
                     </div>
                     <div>
                         Secure Phone
-                        <input autoComplete='tel' id={"securePhone"} value={this.props.value.securePhone || ''} className="form-control" required type="text" dir="auto" onChange={this.props.handleChange}/>
+                        <input autoComplete='tel' id={"securePhone"} value={this.props.value.securePhone || ''} className="form-control" required type="tel" dir="auto" pattern="^(\d{3})?[- .]?\d{3}[- .]?\d{4}?((x|ext|extension)\d+)?$" title="Format: ###-###-#### or ###-###-####x####"  onChange={this.props.handleChange}/>
+                        <div className="errorText">{this.props.error.securePhone? "Required / ###-###-####x####" : ""}</div>
```
Comment (blynch11p) [inline src/components/createEdit/sections/OwnerContact.js]: Don't make the extension look like it's required.

**Comment by blynch11p** [comment]
I just filled out the entire form, hit save, and got 
```
TypeError: this.props.error is undefined[Learn More]  app.bundle.js:159243:1
"The above error occurred in the <POCForm> component:
```

I'll run through the test again, but I don't recall adding anything strange.

**Comment by blynch11p** [comment]
Went back to the same listing, tried to hit save, and it made me re-add each of the images (Icon, Small Banner, Large Banner)

**Comment by blynch11p** [comment]
Duplicate doesn't work either anymore.

**Comment by flesher1** [comment]
Everything should be resolved.

**Comment by blynch11p** [comment]
Seems to be working :+1:

**Comment by irtiza-rizvi** [comment]
works :+1: 

maybe in the future, we can use blueprint's alert instead of the standard JS alert for a nicer look and feel: http://blueprintjs.com/docs/v2/#core/components/alert

**Pull request merged**